crates/ty/tests/cli.rs

```diff
@@ -1062,8 +1061,74 @@ fn config_file_broken_python_setting() -> anyhow::Result<()> {
     WARN ty is pre-release software and not ready for production use. Expect to encounter bugs, missing features, and fatal errors.
     ty failed
       Cause: Invalid search path settings
-      Cause: Failed to discover the site-packages directory: Invalid `--python` argument: `<temp_dir>/not-a-directory-or-executable` does not point to a Python executable or a directory on disk
-    ");
+      Cause: Failed to discover the site-packages directory: Invalid `environment.python` setting in your configuration file `<temp_dir>/pyproject.toml`
+      |
+    1 |
+    2 | [tool.ty.environment]
+    3 | python = "not-a-directory-or-executable"
+      |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ does not point to a Python executable or a directory on disk
+      |
+    "#);
+
+    let case_2 = TestCase::with_files([
+        (
+            "pyproject.toml",
+            r#"
+            [tool.ty.environment]
+            python = "directory-but-no-site-packages"
+            "#,
+        ),
+        ("directory-but-no-site-packages/foo.py", ""),
+        ("test.py", ""),
+    ])?;
+
+    assert_cmd_snapshot!(case_2.command(), @r#"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    WARN ty is pre-release software and not ready for production use. Expect to encounter bugs, missing features, and fatal errors.
+    ty failed
+      Cause: Invalid search path settings
+      Cause: Failed to discover the site-packages directory: Failed to iterate over the contents of the `lib` directory of the Python installation
+      |
```

Comment:
We should display the file name here. It's otherwise unclear where this snippet comes from. E.g. I don't understand what the message shows here.